File: day1/python/day1.py
import os
os.chdir("..")
path = os.getcwd()
data = open(path+"/input.txt", "r")


def calories(data: str):
    elves = data.split("\n\n")
    idk = []
    for i in elves:
        idk.append(sum([int(j) for j in i.split("\n")]))
    return idk


def stage1(list: list):
    highest = list[0]
    for i in list:
        if i > highest:
            highest = i
    return highest


def stage2(list: list):
    vector = []
    for i in range(3):
        highest = stage1(list)
        vector.append(highest)
        list.remove(highest)
    return sum(vector)


parsed = calories(data.read())
data.close()
print("stage 1 = "+str(stage1(parsed)))
print("stage 2 = "+str(stage2(parsed)))

File: day1/golang/day1.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

func input() []int {
	file, _ := os.ReadFile("../input.txt")
	class := strings.Split(string(file), "\n\n")
	var ration []int
	for _, iter := range class {
		total := 0
		for _, j := range strings.Split(iter, "\n") {
			num, _ := strconv.Atoi(j)
			total += num
		}
		ration = append(ration, total)
	}
	return ration
}

func removeList(li []int, i int) []int {
	li[i] = li[len(li)-1]
	return li[:len(li)-1]
}

func highest(in []int) (int, int) {
	highest := in[0]
	index := 0
	for i, j := range in {
		if j > highest {
			highest = j
			index = i
		}
	}
	return index, highest
}

func main() {
	myInput := input()
	_, stage1 := highest(myInput)
	fmt.Println("stage 1 = ", stage1)
	stage2 := 0
	for i := 0; i < 3; i++ {
		hind, val := highest(myInput)
		myInput = removeList(myInput, hind)
		stage2 += val
	}
	fmt.Print("stage 2 = ", stage2)
}
